core/src/main/scala/cats/syntax/either.scala

```diff
@@ -189,9 +254,13 @@ final class EitherOps[A, B](val eab: Either[A, B]) extends AnyVal {
    * }}}
    */
   def toEitherT[F[_]: Applicative]: EitherT[F, A, B] = EitherT.fromEither(eab)
+
+  private[cats] def leftCast[C]: Either[C, B] = eab.asInstanceOf[Either[C, B]]
+
+  private[cats] def rightCast[C]: Either[A, C] = eab.asInstanceOf[Either[A, C]]
```

Comment:
I'm personally not crazy about this. I would rather the following safe methods:
```scala
  def leftCast[A, B, C](l: Left[A, B]): Either[A, C] = l.asInstanceOf[Left[A, C]]
  def rightCast[A, B, C](l: Right[A, B]): Either[C, B] = l.asInstanceOf[Right[C, B]]
```
the ones we have here are only safe when applied to the right `right/left` type. The above is at least type-checked.